### PR TITLE
[SYCL][E2E] Fix deprecated warnings in `Config` tests

### PR DESCRIPTION
**Pull request by ayylol**

### CODE REVIEW

File: sycl/test-e2e/Config/device_selector.cpp [hunk 1]
```diff
@@ -8,20 +8,19 @@
...
+  auto RejectEverything = [](const sycl::device &) { return -1; };
   try {
-    sycl::device Device(Selector);
-  } catch (sycl::runtime_error &E) {
-    return 0;
+    sycl::device Device(RejectEverything);
+  } catch (sycl::exception &E) {
+    if (E.code() == sycl::errc::runtime &&
+        std::string(E.what()).find("No device of requested type available. -1 "
+                                   "(PI_ERROR_DEVICE_NOT_FOUND)") !=
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Config/device_selector.cpp:18]: I'm updating SYCL exception to align with SYCL 2020 spec and as part of it I'm going to remove the `PI_ERROR_DEVICE_NOT_FOUND` part of the message. The reason is `PI_*` are internals of our implementation and not something standard/common in the industry. The closed thing would be OpenCL/L0 error codes, but those are different (logically, not essentially bitwise different) from PI/UR error codes. Backend error codes could be accessed from the `sycl::exception` in a backend specific way defined by the corresponding interop spec, but I don't think it's worth checking that in this test at all. The first part of the condition (`errc::runtime`) and possibly "No device of request type available." substring should suffice.

File: sycl/test-e2e/Config/env_vars.cpp [hunk 2]
```diff
@@ -29,12 +28,10 @@ int main() {
         auto B = buf.get_access<access::mode::read_write>(cgh);
         cgh.single_task<class kernel_name1>([=]() { B[0] = 0; });
       });
-    } catch (sycl::runtime_error &e) {
+    } catch (sycl::exception &e) {
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Config/env_vars.cpp:31]: I have a WIP patch that does this instead:
```
--- a/sycl/test-e2e/Config/env_vars.cpp
+++ b/sycl/test-e2e/Config/env_vars.cpp
@@ -33,8 +33,9 @@ int main() {
       // Exit immediately, otherwise the buffer destructor may actually try to
       // enqueue the command once again, and throw another exception.
       exit(0);
-    } catch (sycl::compile_program_error &e) {
-      exit(0);
+    } catch (sycl::exception &e) {
+      if (e.code() == errc::build)
+        exit(0);
     }
     assert(0 && "Expected exception was *not* thrown");
   } else {
```

I'm only changing one exception and not the other because I'm removing them one by one. I think you can either revert the change or check for both `errc::build` and `errc::runtime` error codes (and fail if not one of the two).

**Comment by againull** [comment]
submit_time.cpp failure is unrelated and reported here: https://github.com/intel/llvm/issues/13591
